pkg/dnfjson/dnfjson.go

```diff
@@ -721,6 +721,13 @@ func (err Error) Error() string {
 // message.
 func parseError(data []byte, repos []repoConfig) Error {
 	var e Error
+	if len(data) == 0 {
```

Comment:
I would love to see at least a tiny bit of testing, maybe something like:
```go
diff --git a/pkg/dnfjson/dnfjson_test.go b/pkg/dnfjson/dnfjson_test.go
index 20771ce5a..13e2effe0 100644
--- a/pkg/dnfjson/dnfjson_test.go
+++ b/pkg/dnfjson/dnfjson_test.go
@@ -776,3 +776,15 @@ func TestRepoConfigMarshalAlsmostEmpty(t *testing.T) {
        // double check here that anything that uses pointers has "omitempty" set
        assert.Equal(t, string(js), `{"id":"","gpgcheck":false,"repo_gpgcheck":false}`)
 }
+
+func TestRunErrorEmptyOutput(t *testing.T) {
+       fakeDnfJsonPath := filepath.Join(t.TempDir(), "dnfjson")
+       fakeDnfJsonNoOutput := `#!/bin/sh -e
+exit 1
+`
+       err := os.WriteFile(fakeDnfJsonPath, []byte(fakeDnfJsonNoOutput), 0o755)
+       assert.NoError(t, err)
+
+       _, err = run([]string{fakeDnfJsonPath}, &Request{})
+       assert.EqualError(t, err, `DNF error occurred: InternalError: dnf-json output was empty`)
+}
```
so that if we need to tweak our strings we can easily check that they are correct. Go error handling has this slightly annyoing tendency to "stutter" as it is often hard to know how much context the previous layer provided. The only good way to deal with this I found so far was just writing tests :/ But I think it's a bit excessive to test all of the new errors, but having an example might be good to double check for "stuttering" and also that the formating looks nice.